components/filter/test/filter.vdiff.js

```diff
@@ -0,0 +1,269 @@
+			elem.opened = true;
+			await oneEvent(elem, 'd2l-filter-dimension-first-open');
+			await nextFrame();
+			await expect(elem).to.be.golden();
+		});
+
+		describe('searched', () => {
+			[
+				{ name: 'single-selection', search: 'empty', template: createSingleDimSingleSelection() },
+				{ name: 'single-selection-select-all', search: 'w', template: createSingleDimSingleSelection({ selected: true }) },
+				{ name: 'multi-selection', search: 'empty', template: createSingleDimWithCounts() },
+				{ name: 'multi-selection-header-text', search: 'a', template: createSingleDimWithCounts({ short: true, headerText: 'Related Courses at Your Company' }) },
+				{ name: 'multi-selection-selected-first', search: 'a', template: createSingleDimWithCounts({ selectedFirst: true }) },
+				{ name: 'multi-selection-clamping', search: 'st', template: createSingleDim({ selected: true, selectAll: true, clampingValues: true }) }
+			].forEach(({ name, template, search }) => {
+				it(name, async() => {
+					const elem = await fixture(template);
+					elem._handleSearch({ detail: { value: search } });
+					await elem.updateComplete;
+					elem.opened = true;
+					await oneEvent(elem, 'd2l-filter-dimension-first-open');
+					await nextFrame();
+					await expect(elem).to.be.golden();
+				});
+			});
+		});
+
+		[
+			{ name: 'press-clear', allSelected: true, selector: 'd2l-button-subtle' },
+			{ name: 'press-unselect-all', allSelected: true, selector: 'd2l-selection-select-all', openUp: true },
+			{ name: 'press-select-all', allSelected: false, selector: 'd2l-selection-select-all', openUp: true }
+		].forEach(({ name, allSelected, selector, openUp }) => {
+			it(name, async() => {
+				const elem = await fixture(createFilterWrapper(html`
+					<d2l-filter-dimension-set key="course" text="Course" select-all>
+						<d2l-filter-dimension-set-value key="art" text="Art" ?selected="${allSelected}"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="astronomy" text="Astronomy" ?selected="${allSelected}"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="biology" text="Biology" ?selected="${allSelected}"></d2l-filter-dimension-set-value>
+					</d2l-filter-dimension-set>
+				`, { openUp }));
+				elem.opened = true;
+				await oneEvent(elem, 'd2l-filter-dimension-first-open');
+				await clickElem(elem.shadowRoot.querySelector(selector));
+				await expect(elem).to.be.golden();
+			});
+		});
+	});
+
+	describe('multiple', () => {
+		async function enterDimension(filter, dimNum) {
+			const menu = filter.shadowRoot.querySelector('d2l-menu');
+			const initialMenuHeight = menu.clientHeight;
+			const dims = filter.shadowRoot.querySelectorAll('d2l-menu-item');
+			sendKeysElem(dims[dimNum - 1], 'press', 'Enter');
+			await oneEvent(filter, 'd2l-filter-dimension-first-open');
+			await waitUntil(() => menu.clientHeight !== initialMenuHeight);
+		}
```

Comment:
This was giving me a lot of trouble, but I think I've found a good combination of flake-resistant but flexible to future changes

Reply:
Seems like it was painful for sure. This is the kind of thing we could put into a "menu test SDK" or something if it continues to crop up.